src/herder/test/HerderTests.cpp

```diff
@@ -1617,6 +1617,206 @@ TEST_CASE("SCP State", "[herder][acceptance]")
     }
 }
 
+TEST_CASE("values externalized out of order", "[herder][quickRestart]")
+{
+    auto networkID = sha256(getTestConfig().NETWORK_PASSPHRASE);
+    auto simulation =
+        std::make_shared<Simulation>(Simulation::OVER_LOOPBACK, networkID);
+
+    auto validatorKey = SecretKey::fromSeed(sha256("validator"));
+    auto listenerKey = SecretKey::fromSeed(sha256("listener"));
+
+    SCPQuorumSet qset;
+    qset.threshold = 1;
+    qset.validators.push_back(validatorKey.getPublicKey());
+
+    auto cfg1 = getTestConfig(1);
+    auto cfg2 = getTestConfig(2);
+    cfg1.MAX_SLOTS_TO_REMEMBER = 5;
+    // Setup a fake archive to prevent HistoryManager from throwing
+    cfg2.HISTORY["test"] = HistoryArchiveConfiguration{"test"};
```

Comment:
why do we need this fake archive? I don't think we set this in other simulations

Reply:
You're right, we do not actually need it (catchup is never started, so no need to set up the archive).